api_core/tests/unit/test_page_iterator.py

```diff
@@ -69,6 +69,22 @@ def test_iterator_calls_parent_item_to_value(self):
         item_to_value.assert_called_with(parent, 12)
         assert page.remaining == 97
 
+    def test_raw_page(self):
+        parent = mock.sentinel.parent
+        item_to_value = mock.sentinel.item_to_value
+
+        raw_page = mock.sentinel.raw_page
+
+        page = page_iterator.Page(parent, (1, 2, 3), item_to_value, raw_page)
+        assert page.raw_page is raw_page
+
+        try:
+            page.raw_page = None
+        except AttributeError:
+            pass
+        else:
+            assert False, "The raw_page attribute should be a read-only property"
```

Comment:
Rather than the `assert False`, use `assertRaises`, e.g.:

```python
        with self.assertRaises(AttributeError):
            page.raw_page = None
```

Reply:
Doh! Don't know what I was thinking.